src/pocket/PocketSynthetic.ts

```diff
@@ -0,0 +1,224 @@
+import { Resource } from 'cdktf';
+import { Synthetics, IAM, CloudWatch } from '@cdktf/provider-aws';
+import { Construct } from 'constructs';
+import { ArchiveProvider, DataArchiveFile } from '@cdktf/provider-archive';
+
+// the exported type is called compile? I assume
+// compile is a builtin?
+import comile = require('string-template/compile')
+import fs = require('fs');
+import path = require('path');
+   
+
+export interface PocketSyntheticProps {
+  config?: {
+    url: string,
+    path: string;
+    port: string;
+    protocol: string;
+  },
+  check?: any;
+  environment: string;
+  artifactS3Location: string;
+  pagerDutyArn?: string;
+}
+
+const defaultMonitor = comile(`
+var synthetics = require('Synthetics');
+
+const apiCanary = async function () {
+
+    // Handle validation for positive scenario
+    const validateSuccessfull = async function (res) {
+        return new Promise((resolve, reject) => {
+            if (res.statusCode < 200 || res.statusCode > 299) {
+                throw res.statusCode + ' ' + res.statusMessage;
+            }
+
+            let responseBody = '';
+            res.on('data', (d) => {
+                responseBody += d;
+            });
+
+            res.on('end', () => {
+                // Add validation on 'responseBody' here if required.
+                resolve();
+            });
+        });
+    };
+
+    let commonProps = {
+        hostname: '{0}',
+        method: 'GET',
+        protocol: '{1}',
+        port: '{2}',
+        region: process.env.AWS_REGION,
+        body:  "",
+        headers: {
+            "User-Agent": synthetics.getCanaryUserAgentString(),
+        },
+    }
+
+    let allConfig = {
+        includeRequestHeaders: true,
+        includeResponseHeaders: true,
+        includeRequestBody: true,
+        includeResponseBody: false,
+        restrictedHeaders: ["authorization", "x-amz-security-token"],
+        continueOnHttpStepFailure: true
+    };
+
+    await synthetics.executeHttpStep('Verify synthetic check', {
+        ...commonProps,
+        path: '{3}',
+    }, validateSuccessfull, allConfig);
+};
+
+exports.handler = async () => {
+    return await apiCanary();
+};
+`);
+
+export class PocketSyntheticCheck extends Resource {
+  constructor(
+    scope: Construct,
+    private name: string,
+    private config: PocketSyntheticProps
+  ) {
+    super(scope, name);
+    new ArchiveProvider(this, 'archive', {});
+   
+    if (!this.config.hasOwnProperty('check')) {
+      this.config.check = defaultMonitor(
+          this.config.config.url,
+          this.config.config.protocol,
+          this.config.config.port,
+          this.config.config.path,
+        ) 
+    }
+
+    const indexPath = path.join(__dirname, 'synthetics', 'nodejs',
+        'node_modules', 'index.js');
+
+    fs.writeFile(indexPath, this.config.check, err => {
```

Comment:
suggestion: prefer `fs.writeFileSync` here to ensure the path is created before zipping the directory on like `110`